OrbitLinuxTracing/PerfEventRingBuffer.cpp

```diff
@@ -121,8 +121,10 @@ void PerfEventRingBuffer::SkipRecord(const perf_event_header& header) {
   WriteRingBufferTail(metadata_page_, new_tail);
 }
 
-void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record) {
-  ReadAtTail(static_cast<uint8_t*>(record), header.size);
+void PerfEventRingBuffer::ConsumeRecord(const perf_event_header& header, void* record,
+                                        size_t size) {
+  DCHECK(header.size == size);
+  ReadAtTail(static_cast<uint8_t*>(record), size);
```

Comment:
Now `size` is passed just for the (`D`)`CHECK`. Can we just remove it and use the implementation we had before, with the difference that now it is not called from outside (the public method is templated and does the check, and this check is now redundant)?

Reply:
And also maybe just rename the private method to `ConsumeRecordInternal` or something like that to avoid confusion with overloading.

Reply:
So again, can we just remove the `size` argument now as it's only used for the `DCHECK`, which is redundant as this is a private method and the caller already does the `CHECK`?

Reply:
I would like to keep it here.

Reply:
Ok, then use a normal `CHECK` otherwise it's useless, the service is run in debug mode too rarely (performance reason + it's hard to debug on an instance).

Reply:
well, on second look it does look wired. I guess I will add a comment here and remove the check and size argument.